core-backend/src/funcs.rs

```diff
@@ -675,6 +675,45 @@ where
         })
     }
 
+    pub fn free_range(start: u32, end: u32) -> impl SysCall<Ext, i32> {
+        let page_count = start.abs_diff(end).saturating_add(1);
+
+        InfallibleSysCall::new(
+            RuntimeCosts::FreeRange(page_count),
+            move |ctx: &mut CallerWrap<Ext>| {
+                if start > end {
+                    log::trace!("Invalid range {start:?}:{end:?}");
+                    return Err(UndefinedTerminationReason::Actor(
+                        ActorTerminationReason::Trap(TrapExplanation::Unknown),
```

Comment:
I'm not sure `Unknown` is a good choice here. The error is pretty clear - invalid range. Maybe introduce a new variant in `AllocError` and move that check to the `ext.rs`?

what do you think? @grishasobol @holykol

Reply:
done! See AllocError::InvalidRange